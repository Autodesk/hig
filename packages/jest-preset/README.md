# @hig/jest-preset

Jest preset for HIG packages.

## Usage

_Example `package.json`:_

```json
{
  "jest": {
    "preset": "@hig/jest-preset"
  }
}
```
