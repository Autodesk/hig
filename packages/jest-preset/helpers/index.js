/* eslint-disable global-require */
module.exports = {
  takeSnapshotsOf: require("./takeSnapshotsOf")
};
