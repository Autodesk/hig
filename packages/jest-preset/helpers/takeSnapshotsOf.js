const React = require("react");
const renderer = require("react-test-renderer");

module.exports = function takeSnapshotsOf(Component, cases, rendererOptions) {
  cases.forEach(({ desc, props }) => {
    it(desc, () => {
      const vDom = React.createElement(Component, props);
      const tree = renderer.create(vDom, rendererOptions).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
};
