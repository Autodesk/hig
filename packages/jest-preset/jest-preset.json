{
  "collectCoverageFrom": ["**/src/**/*.{js,jsx}"],
  "coverageDirectory": "<rootDir>/coverage",
  "coveragePathIgnorePatterns": [
    "__gemini__",
    "__stories__",
    "docs",
    "examples/redux",
    "node_modules",
    "packages/jest-preset",
    "packages/vanilla",
    "src/playground"
  ],
  "coverageThreshold": {
    "global": {
      "branches": 50,
      "functions": 50,
      "lines": 50,
      "statements": 50
    }
  },
  "moduleNameMapper": {
    "\\.(css|scss|svg)$": "@hig/jest-preset/fileMock.js"
  },
  "modulePaths": ["node_modules"],
  "moduleFileExtensions": ["js", "jsx", "json"],
  "setupFiles": ["raf/polyfill"],
  "setupTestFrameworkScriptFile": "@hig/jest-preset/setupTests.js",
  "testPathIgnorePatterns": [
    "docs",
    "packages/babel-preset",
    "packages/jest-preset",
    "packages/vanilla"
  ],
  "unmockedModulePathPatterns": ["node_modules"]
}
