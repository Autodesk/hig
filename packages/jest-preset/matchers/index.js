require("./toBeConstants");
require("./toImplementHIGInterfaceOf");
require("./toHavePropTypesAndDocgenInfo");
