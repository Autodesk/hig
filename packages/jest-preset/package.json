{
  "private": true,
  "name": "@hig/jest-preset",
  "version": "0.1.0",
  "devDependencies": {
    "babel-core": "^6.26.3",
    "babel-jest": "^23.4.0",
    "enzyme": "^2.9.1",
    "jest": "^21.2.1",
    "jest-enzyme": "^3.8.3",
    "jest-extended": "^0.6.0",
    "raf": "^3.4.0",
    "react-test-renderer": "^15.6.1"
  },
  "eslintConfig": {
    "extends": "@hig"
  },
  "jest": {
    "preset": "@hig/jest-preset"
  }
}
