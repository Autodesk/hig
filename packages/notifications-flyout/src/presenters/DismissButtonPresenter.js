import React from "react";
import PropTypes from "prop-types";
import { css, cx } from "emotion";
import IconButton from "@weave-design/icon-button";
import { CloseSUI } from "@weave-design/icons";
import { createCustomClassNames } from "@weave-design/utils";

import stylesheet from "./stylesheet";

export default function DismissButtonPresenter({
  hasHover,
  onClick,
  stylesheet: customStylesheet,
  title,
  ...otherProps
}) {
  const { className } = otherProps;
  const dismissButtonClassName = createCustomClassNames(
    className,
    "dismiss-button"
  );
  const styles = stylesheet({ hasHover, stylesheet: customStylesheet }, {});
  return (
    <div className={cx([dismissButtonClassName, css(styles.dismissButton)])}>
      <IconButton onClick={onClick} icon={<CloseSUI />} title={title} />
    </div>
  );
}

DismissButtonPresenter.defaultProps = {
  title: "Dismiss featured notification",
};

DismissButtonPresenter.propTypes = {
  hasHover: PropTypes.bool,
  onClick: PropTypes.func,
  stylesheet: PropTypes.func,
  title: PropTypes.string,
};
