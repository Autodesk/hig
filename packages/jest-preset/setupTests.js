require("jest-enzyme");
require("jest-extended");
require("./matchers");
