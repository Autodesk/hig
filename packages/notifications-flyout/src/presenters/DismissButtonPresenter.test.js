import { takeSnapshotsOf } from "@weave-design/jest-preset/helpers";

import DismissButtonPresenter from "./DismissButtonPresenter";

describe("notifications-flyout/presenters/DismissButtonPresenter", () => {
  takeSnapshotsOf(DismissButtonPresenter, [
    {
      desc: "renders without props",
      props: {},
    },
    {
      desc: "renders with all props",
      props: {
        onClick: () => {},
        title: "HELLO",
      },
    },
  ]);
});
