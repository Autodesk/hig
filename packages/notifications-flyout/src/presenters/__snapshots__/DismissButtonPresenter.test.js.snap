// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notifications-flyout/presenters/DismissButtonPresenter renders with all props 1`] = `
.emotion-2 {
  display: none;
  position: absolute;
  top: 0;
  right: 0;
}

.emotion-1 {
  border: 2px solid transparent;
  display: inline-block;
  position: relative;
  cursor: pointer;
  box-sizing: border-box;
  border-radius: 2px;
  padding: 0;
  height: calc(20px + (8px * 2));
  line-height: calc(20px + (8px * 2));
  width: calc(20px + (8px * 2));
  outline: 0;
  -webkit-transition-property: box-shadow,background-color;
  transition-property: box-shadow,background-color;
  -webkit-transition-duration: 0.3s,0.3s;
  transition-duration: 0.3s,0.3s;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1),cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1),cubic-bezier(0.4,0,0.2,1);
  box-shadow: none;
  background-color: transparent;
}

.emotion-1 svg * {
  fill: #808080;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: fill;
  transition-property: fill;
}

.emotion-0 {
  fill: #808080;
  pointer-events: none;
  position: absolute;
  left: 0;
  right: 0;
  margin: 0 auto;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  pointer-events: none;
}

.emotion-0 > * {
  fill: #808080;
  pointer-events: none;
}

<div
  className="emotion-2"
>
  <button
    className="emotion-1"
    disabled={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    tabIndex="0"
    title="HELLO"
  >
    <svg
      className="emotion-0"
      height="16px"
      viewBox="0 0 16 16"
      width="16px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        close-s
      </title>
      <path
        d="M13.4 3.4l-.8-.8L8 7.3 3.4 2.6l-.8.8L7.3 8l-4.7 4.6.8.8L8 8.7l4.6 4.7.8-.8L8.7 8l4.7-4.6z"
      />
    </svg>
  </button>
</div>
`;

exports[`notifications-flyout/presenters/DismissButtonPresenter renders without props 1`] = `
.emotion-2 {
  display: none;
  position: absolute;
  top: 0;
  right: 0;
}

.emotion-1 {
  border: 2px solid transparent;
  display: inline-block;
  position: relative;
  cursor: pointer;
  box-sizing: border-box;
  border-radius: 2px;
  padding: 0;
  height: calc(20px + (8px * 2));
  line-height: calc(20px + (8px * 2));
  width: calc(20px + (8px * 2));
  outline: 0;
  -webkit-transition-property: box-shadow,background-color;
  transition-property: box-shadow,background-color;
  -webkit-transition-duration: 0.3s,0.3s;
  transition-duration: 0.3s,0.3s;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1),cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1),cubic-bezier(0.4,0,0.2,1);
  box-shadow: none;
  background-color: transparent;
}

.emotion-1 svg * {
  fill: #808080;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: fill;
  transition-property: fill;
}

.emotion-0 {
  fill: #808080;
  pointer-events: none;
  position: absolute;
  left: 0;
  right: 0;
  margin: 0 auto;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  pointer-events: none;
}

.emotion-0 > * {
  fill: #808080;
  pointer-events: none;
}

<div
  className="emotion-2"
>
  <button
    className="emotion-1"
    disabled={undefined}
    onBlur={[Function]}
    onClick={undefined}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    tabIndex="0"
    title="Dismiss featured notification"
  >
    <svg
      className="emotion-0"
      height="16px"
      viewBox="0 0 16 16"
      width="16px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        close-s
      </title>
      <path
        d="M13.4 3.4l-.8-.8L8 7.3 3.4 2.6l-.8.8L7.3 8l-4.7 4.6.8.8L8 8.7l4.6 4.7.8-.8L8.7 8l4.7-4.6z"
      />
    </svg>
  </button>
</div>
`;
